src/middlewared/middlewared/pytest/unit/utils/test_mountinfo.py

```diff
@@ -0,0 +1,89 @@
+453 446 0:74 / /mnt/dozer/administrative_share/backups_dataset/userdata/DOMAIN_GOAT rw,noatime shared:253 - zfs dozer/administrative_share/backups_dataset/userdata/DOMAIN_GOAT rw,xattr,posixacl
+460 453 0:75 / /mnt/dozer/administrative_share/backups_dataset/userdata/DOMAIN_GOAT/bob rw,noatime shared:257 - zfs dozer/administrative_share/backups_dataset/userdata/DOMAIN_GOAT/bob rw,xattr,posixacl
+306 292 0:53 / /mnt/dozer/EXPORT rw,noatime shared:169 - zfs dozer/EXPORT rw,xattr,posixacl
+362 292 0:60 / /mnt/dozer/noacl rw,noatime shared:201 - zfs dozer/noacl rw,xattr,noacl
+93 26 0:38 / /var/db/system rw,relatime shared:48 - zfs dozer/.system rw,xattr,noacl
+100 93 0:39 / /var/db/system/cores rw,relatime shared:52 - zfs dozer/.system/cores rw,xattr,noacl
+107 93 0:40 / /var/db/system/samba4 rw,relatime shared:56 - zfs dozer/.system/samba4 rw,xattr,noacl
+114 93 0:41 / /var/db/system/syslog-f803551cf3cd4df8a1ddb0569466b72a rw,relatime shared:60 - zfs dozer/.system/syslog-f803551cf3cd4df8a1ddb0569466b72a rw,xattr,noacl
+121 93 0:42 / /var/db/system/rrd-f803551cf3cd4df8a1ddb0569466b72a rw,relatime shared:64 - zfs dozer/.system/rrd-f803551cf3cd4df8a1ddb0569466b72a rw,xattr,noacl
+151 93 0:43 / /var/db/system/configs-f803551cf3cd4df8a1ddb0569466b72a rw,relatime shared:68 - zfs dozer/.system/configs-f803551cf3cd4df8a1ddb0569466b72a rw,xattr,noacl
+158 93 0:44 / /var/db/system/webui rw,relatime shared:100 - zfs dozer/.system/webui rw,xattr,noacl
+187 93 0:45 / /var/db/system/services rw,relatime shared:104 - zfs dozer/.system/services rw,xattr,noacl
+234 93 0:46 / /var/db/system/glusterd rw,relatime shared:143 - zfs dozer/.system/glusterd rw,xattr,noacl
+241 93 0:47 / /var/db/system/ctdb_shared_vol rw,relatime shared:147 - zfs dozer/.system/ctdb_shared_vol rw,xattr,noacl
+271 26 0:39 / /var/lib/systemd/coredump rw,relatime shared:52 - zfs dozer/.system/cores rw,xattr,noacl
+467 26 0:76 / /mnt/tank\040space\040 rw,noatime shared:261 - zfs tank\040space\040 rw,xattr,posixacl
+474 467 0:77 / /mnt/tank\040space\040/Dataset\040With\040a\040space rw,noatime shared:265 - zfs tank\040space\040/Dataset\040With\040a\040space rw,xattr,posixacl
+"""
+
+
+def test__mntinfo_spaces():
+    line = r'474 467 0:77 / /mnt/tank\040space\040/Dataset\040With\040a\040space rw,noatime shared:265 - zfs tank\040space\040/Dataset\040With\040a\040space rw,xattr,posixacl'
+    data = {}
+    __parse_mntent(line, data)
+    assert 77 in data
+    mntent = data[77]
+    assert mntent['mount_id'] == 474
+    assert mntent['parent_id'] == 467
+    assert mntent['device_id'] == {'major': 0, 'minor': 77, 'dev_t': 77}
+    assert mntent['root'] == '/'
+    assert mntent['mountpoint'] == '/mnt/tank space /Dataset With a space'
+    assert mntent['mount_opts'] == ['RW', 'NOATIME']
+    assert mntent['fs_type'] == 'zfs'
+    assert mntent['mount_source'] == 'tank space /Dataset With a space'
+    assert mntent['super_opts'] == ['RW', 'XATTR', 'POSIXACL']
+
+
+def test__getmntinfo():
```

Comment:
Would be nice if we tested the actual output of the contents as well to ensure that they actually match to what we are expecting and not just that we were able to parse the fake file.